### PR TITLE
Rename teams to team plans

### PR DESCRIPTION
**Pull request by gtsiolis**
## Description

This will rename the current _Teams_ feature to _Team Plans_ to avoid confusion with the upcoming _Teams & Projects_ feature which will introduce a way to create a team and invite members. 🍵 

We've decided to proceed with this MVC (minimum viable change) instead of adding a deprecation notice in https://github.com/gitpod-io/gitpod/pull/5121. 🙀 

There's also a follow up PR to update the relevant docs section in https://github.com/gitpod-io/website/pull/1032.
> https://www.gitpod.io/docs/teams

See relevant discussion points[[1](https://www.notion.so/gitpod/Team-Meta-Sync-3cedd13aa1fb4a0380a7ff1b5c4de971#0698ada4491b4bb1bb593de277545c2e)][[2](https://www.notion.so/gitpod/Team-Meta-Sync-3cedd13aa1fb4a0380a7ff1b5c4de971#d0ad4cf1e16a4eb4b82902e5c752d50a)] (internal).

## How to test

1. Go to `/teams`
2. Create a team plan, manage team members.

## Release Notes
```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **gtsiolis** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/settings/Teams.tsx
Comment (jankeromnes) [inline components/dashboard/src/settings/Teams.tsx:431]:
```suggestion
                <h2>Manage team plans and team members.</h2>
```

File: components/dashboard/src/settings/Teams.tsx
Comment (jankeromnes) [inline components/dashboard/src/settings/Teams.tsx:430]:
```suggestion
                <h3 className="self-center">All Team Plans</h3>
```

File: components/dashboard/src/settings/settings-menu.ts
Comment (gtsiolis) [inline components/dashboard/src/settings/settings-menu.ts:21]: **thought**: Alternatively, this could read _Subscriptions_.

**Review by jankeromnes — APPROVED** [review]
Looks good to me! Many thanks @gtsiolis 🙏

Please see small inline nits, otherwise good to merge 🚀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 1f7c929b50f4ad6a5b741e5f79e6312d96c4e202</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6011#pullrequestreview-770303363" title="Approved">jankeromnes</a>*

Associated issue: *#698*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/settings/settings-menu.ts
Comment (jankeromnes) [reply components/dashboard/src/settings/settings-menu.ts:21]: So, the reason we initially went with "Plans" and not "Subscriptions" for the individual "Plans" page is:

- An individual has only one Subscription with Gitpod (paid or not), but that Subscription can be upgraded/downgraded to various Plans (e.g. Open Source, Professional, Unleashed, etc)

So the page could have been called "Subscription" (singular) or "Plans" (plural). We felt "Plans" more appropriate, because that's exactly what this page shows you (a collection of available Plans).

For Teams, this is a bit more awkward, because you can possibly have up to three Team Subscriptions (one for each Team Plan).

I'm a bit more in favour of "Team Plans" (to be consistent with "Plans"), but I agree that this could also become "Team Subscriptions".

**Comment by jankeromnes** [comment]
Whoops, forgot about auto-merge... 😅 Anyway, please feel free to address the nits in a follow-up PR. Thanks again!

**Comment by gtsiolis** [comment]
Oh, this got merged early! 🙀 

Added follow-up PR in https://github.com/gitpod-io/gitpod/pull/6013.

Thanks for taking a look at this, @jankeromnes!

**Pull request merged**
